commercetools.Sdk/commercetools.Sdk.HttpApi.IntegrationTests/ProductProjections/ProductProjectionsIntegrationTests.cs

```diff
@@ -96,25 +97,113 @@ public void QueryAndOffsetStagedProductProjections()
             //Arrange
             IClient commerceToolsClient = this.productProjectionsFixture.GetService<IClient>();
 
+            ProductType productType = this.productProjectionsFixture.productFixture.CreateNewProductType();
+
             for (int i = 0; i < 3; i++)
             {
-                Product stagedProduct = this.productProjectionsFixture.productFixture.CreateProduct(true);
+                Product stagedProduct = this.productProjectionsFixture.productFixture.CreateProduct(productType, true);
                 this.productProjectionsFixture.productFixture.ProductsToDelete.Add(stagedProduct);
             }
 
             //Act
             ProductProjectionAdditionalParameters stagedAdditionalParameters = new ProductProjectionAdditionalParameters();
             stagedAdditionalParameters.Staged = true;
 
-            QueryPredicate<ProductProjection> queryPredicate = new QueryPredicate<ProductProjection>(productProjection => productProjection.Version == 1);
+            //Retrieve Products with created productType
             QueryCommand<ProductProjection> queryCommand = new QueryCommand<ProductProjection>(stagedAdditionalParameters);
-            queryCommand.SetWhere(queryPredicate);
+            queryCommand.Where(productProjection => productProjection.ProductType.Id == productType.Id.valueOf().ToString());
             queryCommand.Offset = 2;
             PagedQueryResult<ProductProjection> returnedSet = commerceToolsClient.ExecuteAsync(queryCommand).Result;
 
             //Assert
             Assert.Single(returnedSet.Results);
             Assert.Equal(3, returnedSet.Total);
         }
+
+        [Fact]
+        public void QueryAndLimitStagedProductProjections()
+        {
+            //Arrange
+            IClient commerceToolsClient = this.productProjectionsFixture.GetService<IClient>();
+
+            ProductType productType = this.productProjectionsFixture.productFixture.CreateNewProductType();
+
+            for (int i = 0; i < 3; i++)
+            {
+                Product stagedProduct = this.productProjectionsFixture.productFixture.CreateProduct(productType, true);
+                this.productProjectionsFixture.productFixture.ProductsToDelete.Add(stagedProduct);
+            }
+
+            //Act
+            ProductProjectionAdditionalParameters stagedAdditionalParameters = new ProductProjectionAdditionalParameters();
+            stagedAdditionalParameters.Staged = true;
+
+            //Retrieve Products with created productType limited by 2
+            QueryCommand<ProductProjection> queryCommand = new QueryCommand<ProductProjection>(stagedAdditionalParameters);
+            queryCommand.Where(productProjection => productProjection.ProductType.Id == productType.Id.valueOf().ToString());
```

Comment:
same here with ToString()

Reply:
I used it to avoid this warning "Possible unintended reference comparison; to get a value comparison, cast the right hand side to type 'string'"